### PR TITLE
feat-filters

### PR DESCRIPTION
**Pull request by prathmeshjagtap**
Added Filters to Filter Products
1)Sort By Price Filter
2)Filter By Category
3) Filter by Price Range
4) sort by Price

## Files to  Check 
 
Do check Filter Context, Filter Reducer, Filter Components, and Utils Filter Functions.

- [ ] src/froentend/contexts/filter-context.js
- [ ] src/froentend/pages/products/Filters.jsx
- [ ] src/froentend/reducer/filterReducer.js
- [ ] src/froentend/utils/filters.js

## What not to check

Have added inline style in the home page which will be changed later 
- [ ]  src/froentend/pages/home/Home.jsx

## Live Link: https://deploy-preview-5--bookify-bookstore.netlify.app/   

Go to Products Page from Home Page

### CODE REVIEW

File: src/froentend/pages/home/Home.jsx [hunk 1]
```diff
@@ -5,6 +5,9 @@ function Home() {
 	return (
 		<div>
 			<Navbar />
+			<div style={{ "margin-top": "8rem", textAlign: "center" }}>
```
Comment (Prabhav-Sharma) [inline src/froentend/pages/home/Home.jsx:8]: Avoid inline styling

File: src/index.js [hunk 2]
```diff
@@ -4,14 +4,17 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
+import { FilterProvider } from "./froentend/contexts";
```
Comment (Prabhav-Sharma) [inline src/index.js]: Fix "frontend" typo in the folder structure

**Review by ishapatell1 — APPROVED** [review]

File: src/froentend/utils/filters.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		}
+	}
+
+	return flag ? filterByCategoryData : data;
+};
+
+const filterByRatings = (state, data) => {
+	if (state.ratings) {
+		return data.filter((item) => Number(item.rating) >= Number(state.ratings));
+	} else {
```
Comment (Prabhav-Sharma) [inline src/froentend/utils/filters.js]: Can remove the else call

**Review by Prabhav-Sharma — CHANGES_REQUESTED** [review]
Code is clean and well structured, implement the changes I've commented and you're good to merge

File: src/froentend/utils/filters.js (hunk 3 above)
Comment (prathmeshjagtap) [reply src/froentend/utils/filters.js]: yes Done

File: src/index.js (hunk 2 above)
Comment (prathmeshjagtap) [reply src/index.js]: yes fixed the Typo

File: src/froentend/pages/home/Home.jsx (hunk 1 above)
Comment (prathmeshjagtap) [reply src/froentend/pages/home/Home.jsx:8]: yes i have used inline styling just for Preview Purpose I will remove it Later

Comment (Prabhav-Sharma) [reply src/froentend/pages/home/Home.jsx:8]: Oh, then it's alright ^_^

**Review by Prabhav-Sharma — APPROVED** [review]

File: src/frontend/pages/products/Filters.jsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+								type="checkbox"
+								id={category}
+								onChange={() =>
+									dispatch({
+										type: filterAction.FILTER_BY_CATEGORY,
+										payload: category,
+									})
+								}
+							/>
+							<label>{category}</label>
```
Comment (narharikale) [inline src/frontend/pages/products/Filters.jsx]: we can associate `label` with `input` element using id  or we can just wrap input inside the label element

File: src/frontend/pages/products/Products.jsx [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
+import {
+	ComposeFunction,
+	sortbyPriceFunction,
+	filterByPrice,
+	filterByRatings,
+	filterByCategory,
+} from "../../utils/filters";
 import "./products.css";
 
 const axios = require("axios").default;
```
Comment (narharikale) [inline src/frontend/pages/products/Products.jsx:14]: is there any specific reason for not directly importing Axios?

**Review by narharikale — APPROVED** [review]
code is clean ✅ and i liked how you used compose function

File: src/frontend/pages/products/Products.jsx (hunk 5 above)
Comment (prathmeshjagtap) [reply src/frontend/pages/products/Products.jsx:14]: no will change it later

File: src/frontend/pages/products/Filters.jsx (hunk 4 above)
Comment (prathmeshjagtap) [reply src/frontend/pages/products/Filters.jsx]: yes done

**Pull request merged**